openedx/core/djangoapps/credit/exceptions.py

```diff
@@ -56,3 +60,25 @@ class InvalidCreditStatus(CreditApiBadRequest):
     The status is not either "approved" or "rejected".
     """
     pass
+
+
+class InvalidCreditRequest(APIException):
+    """ API request is invalid. """
+    status_code = status.HTTP_400_BAD_REQUEST
+
+
+class UserNotEligibleException(InvalidCreditRequest):
```

Comment:
Agreed.  Seems like the right to do to reduce risk.  But do we have / can we create a ticket to track this clean up task once the rewrite is solidly in place?